sunpy/timeseries/timeseriesbase.py

```diff
@@ -530,19 +533,32 @@ def truncate(self, a, b=None, int=None):
         `~sunpy.timeseries.TimeSeries`
             A new `~sunpy.timeseries.TimeSeries` with only the selected times.
         """
+        is_int = False
         # Evaluate inputs
         # If given strings, then use to create a sunpy.time.timerange.TimeRange
         # for the SunPy text date parser.
         if isinstance(a, str) and isinstance(b, str):
-            a = TimeRange(a, b)
-        if isinstance(a, TimeRange):
+            time_range = TimeRange(a, b)
+            start = time_range.start.datetime
+            end = time_range.end.datetime
+        elif isinstance(a, TimeRange):
             # If we have a TimeRange, extract the values
             start = a.start.datetime
             end = a.end.datetime
         else:
             # Otherwise we already have the values
             start = a
             end = b
+            is_int = True
```

Comment:
should we add input validation to user give only allowed inputs . `Astropy.Time` , `int`

If we use this 
```
a = pd.Timestamp('2011-06-07 06:55:59')
b = pd.Timestamp('2011-06-07 08:55:59')
```
if a , b is within time range code works fine but when not (out of bound timeranges) it cauese error

Reply:
i think if we are using `pandas.Timestamp` in `Timeseries.data.index` then we should add also this format in input parameters.

Reply:
We use it since astropy.time can not be a true pandas index.